src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/EditGeneralDetailsSectionController.java

```diff
@@ -364,26 +394,209 @@ private void populateLinkedCasesConfirmDropdowns(final Model model) {
         .build();
   }
 
+  /**
+   * Handles the GET request for searching linked cases.
+   *
+   * @param userDetails The details of the logged-in user.
+   * @param model       The UI model to add attributes to.
+   * @param session     The HTTP session object.
+   * @return String     The view name for linked cases search.
+   */
+  @GetMapping("/application/summary/linked-cases/search")
+  public String linkedCasesSearchGet(
+      @SessionAttribute(USER_DETAILS) final UserDetail userDetails,
+      final Model model,
+      final HttpSession session) {
+
+    populateLinkedCasesSearchDropdowns(userDetails, model);
+
+    final CaseSearchCriteria caseSearchCriteria = new CaseSearchCriteria();
+    model.addAttribute(CASE_SEARCH_CRITERIA, caseSearchCriteria);
+    session.setAttribute(CASE_SEARCH_CRITERIA, caseSearchCriteria);
+
+    return "application/summary/application-linked-case-search";
+  }
+
+  /**
+   * Handles the POST request for searching linked cases, validates the search criteria,
+   * and processes search results.
+   *
+   * @param activeCase           Active case details from session attribute.
+   * @param caseSearchCriteria   Search criteria model attribute.
+   * @param user                 User details from session attribute.
+   * @param linkedCases          Linked cases from session attribute.
+   * @param redirectAttributes   Redirect attributes.
+   * @param bindingResult        Binding result for validation.
+   * @param model                Spring MVC model.
+   * @return The view name for redirection or linked case search view.
+   */
+  @PostMapping("/application/summary/linked-cases/search")
+  public String linkedCasesSearchPost(
+      @SessionAttribute(ACTIVE_CASE) final ActiveCase activeCase,
+      @ModelAttribute(CASE_SEARCH_CRITERIA) final CaseSearchCriteria caseSearchCriteria,
+      @SessionAttribute(USER_DETAILS) final UserDetail user,
+      @SessionAttribute("linkedCases") final ResultsDisplay<LinkedCaseResultRowDisplay> linkedCases,
```

Comment:
"linkedCases" should be a constant

Reply:
and maybe call the variable currentLinkedCases or something, to make it more obvious that it isn't related to the search results

Reply:
amended